src/components/bottom_bar_statistic/aggregate_statistics_store.ts

```diff
@@ -0,0 +1,130 @@
+import { sum } from "../../functions/helper_math";
+import { average, countAny, countNumbers, max, min } from "../../functions/helper_statistical";
+import { positions } from "../../helpers";
+import { Get } from "../../store_engine";
+import { SpreadsheetStore } from "../../stores";
+import { _t } from "../../translation";
+import { CellValueType, Command, EvaluatedCell, Locale, isCoreCommand } from "../../types";
+
+export interface StatisticFnResults {
+  [name: string]: number | undefined;
+}
+
+interface SelectionStatisticFunction {
+  name: string;
+  compute: (data: EvaluatedCell[], locale: Locale) => number;
+  types: CellValueType[];
+}
+
+const selectionStatisticFunctions: SelectionStatisticFunction[] = [
+  {
+    name: _t("Sum"),
+    types: [CellValueType.number],
+    compute: (values, locale) => sum([[values]], locale),
+  },
+  {
+    name: _t("Avg"),
+    types: [CellValueType.number],
+    compute: (values, locale) => average([[values]], locale),
+  },
+  {
+    name: _t("Min"),
+    types: [CellValueType.number],
+    compute: (values, locale) => min([[values]], locale),
+  },
+  {
+    name: _t("Max"),
+    types: [CellValueType.number],
+    compute: (values, locale) => max([[values]], locale),
+  },
+  {
+    name: _t("Count"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values) => countAny([[values]]),
+  },
+  {
+    name: _t("Count Numbers"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values, locale) => countNumbers([[values]], locale),
+  },
+];
+
+export class AggregateStatisticsStore extends SpreadsheetStore {
+  statisticFnResults: StatisticFnResults = {};
+  private isDirty = false;
+
+  constructor(get: Get) {
+    super(get);
+    this.model.selection.observe(this, {
+      handleEvent: this.handleEvent.bind(this),
+    });
+  }
+
+  handle(cmd: Command) {
+    if (
+      (isCoreCommand(cmd) && cmd.type !== "UPDATE_CELL") ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd)
```

Comment:
Why all the core commands rather than the commands invalidating the evaluation ? Kinda useless to invalidate the aggregate on an UPATE_LOCALE or ADD_TABLE